core/src/com/unciv/ui/screens/worldscreen/minimap/MinimapTileUtil.kt

```diff
@@ -0,0 +1,28 @@
+package com.unciv.ui.screens.worldscreen.minimap
+
+import com.badlogic.gdx.scenes.scene2d.Group
+import java.awt.geom.Rectangle2D
+import kotlin.math.max
+import kotlin.math.min
+
+object MinimapTileUtil {
+
+    fun spreadOutMinimapTiles(tileLayer: Group, tiles: List<MinimapTile>, tileSize: Float) : Rectangle2D.Float {
+        var topX = -Float.MAX_VALUE
+        var topY = -Float.MAX_VALUE
+        var bottomX = Float.MAX_VALUE
+        var bottomY = Float.MAX_VALUE
+
+        for (image in tiles.map { it.image }) {
+            tileLayer.addActor(image)
+
+            // keeps track of the current top/bottom/left/rightmost tiles to size and position the minimap correctly
+            topX = max(topX, image.x + tileSize)
+            topY = max(topY, image.y + tileSize)
+            bottomX = min(bottomX, image.x)
+            bottomY = min(bottomY, image.y)
+        }
+
+        return Rectangle2D.Float(topX, topY, topX-bottomX, topY-bottomY)
```

Comment:
I suggest replacing with `return Rectangle2D.Float(bottomX , bottomY, topX-bottomX, topY-bottomY)`, because topX/Y is not used, unlike bottomX/Y

Reply:
I understand, however if I read the code right, topX and topY are always greater than bottomX and bottomY respectively, so if I do what you suggest, I'd end up with the lower left corner as the anchor and negative widths and heights just to avoid some computation on the receiver. I think the extra computation is the lesser evil. Let me know if you feel strongly about this. Another option would be to use a custom rectangle that is just defined by two points rather than one point and width and height, but I didn't want to reinvent the wheel for such a basic data structure.

Reply:
topX and topY are not used anywhere except for computing width and height inside function itself. Using a topX/Y as returning arguments, in my opinion, only adds one additional math operation to calculate bottomX/Y, which are used in shifting, at least. So what`s the point?
And Unciv already has 3 different coord layers with different anchor points: hex positioning, hex drawing, stage itself and center, top left, bottom left(maybe another order, I don't remember exactly). 
Anyway, I am not insisting on anything, but only sharing my observations and suggestions. The decision is up to you and @yairm210

Reply:
If we return a point and two lengths (width and height), I agree with you. It would be better to return Point(bottomX, bottomY), width & height, but IMO in the context of a rectangle one expects that you can do otherCornerPoint(xCoord+width, yCoord+height) and the only way to achieve this would be to have negative widths and heights, which I personally find odd. It seems this is supported, but the behavior isn't even documented (https://bugs.openjdk.org/browse/JDK-6800505).

We could introduce a custom wrapper class and not use the rectangle notion. Do you think that would be better or do you have other ideas how to solve this?

Reply:
I'm not sure I understand. bottomX and bottomY are points on the same plane but in a different location. And we can calculate topX and topY with the same formula (bottomX+width, bottomY+height), if we ever need them. 
Am I missing something?

Reply:
```
Rectangle2D.Float(float x, float y, float w, float h)
Constructs and initializes a Rectangle2D from the specified float coordinates.
```

Negative will be its coordinates` x = bottomX` and `y = bottomY`, width will remain unchanged `w = topX-bottomX`, as well as height ` h = topY-bottomY`

e.g we have these parameters:
```
topX = 20
topY = 30
botomX =  -10
botomY= -20
width = 20 - -10 = 30
heigth = 30 - -20 = 50
```
You suggest:
`return (20, 30, 30, 50)`
I suggest:
`return (-10, -20, 30, 50)`

Reply:
Okay, I'm not going to die on this hill. Done! :)